content/javascript/concepts/memoization/memoization.md

```diff
@@ -0,0 +1,75 @@
+---
+Title: 'Memoization'
+Description: 'Memoization can be used to improve performance in JavaScript code.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Algorithms'
+  - 'Conceptual'
+  - 'Dynamic Programming'
+CatalogContent:
+  - 'introduction-to-javascript'
+  - 'paths/computer-science'
+---
+
+In JavaScript, [memoization](https://www.codecademy.com/resources/docs/general/memoization) can be used to improve a program's performance.
+
+This is ideal whenever the same calculations (with the same inputs and outputs) are known to take place. Memoization helps by saving, or caching, these results in computer memory for later use. It can be applied to many contexts, including:
+
+- Making calls to recursive or pure function where some calculations use the same inputs and return the same outputs.
+- Fetching from the server with the same base API endpoints.
+
+## Syntax
+
+```pseudo
+memo = []
+
+function memoFunction(arg) {
+  if(memo[arg]) {
+    return memo[arg]
+  } else {
+    // Do this and update memo.
+  }
+}
+```
+
+Memoized results can be represented with [arrays](https://www.codecademy.com/resources/docs/javascript/arrays) or [objects](https://www.codecademy.com/resources/docs/javascript/objects). In a given function, if a previously calculated result exists in the `memo` dictionary, it is returned. Otherwise, normal calculations take place and the `memo` is updated.
```

Comment:
```suggestion
Memoized results can be represented with [arrays](https://www.codecademy.com/resources/docs/javascript/arrays) or [objects](https://www.codecademy.com/resources/docs/javascript/objects). In a given function, if a previously calculated result exists in the `memo` array, it is returned. Otherwise, normal calculations take place and the `memo` array is updated.
```